### PR TITLE
Added updateSketchCondition as a Draw Interaction option

### PR DESCRIPTION
**Pull request by bjornharrtell**
- [ ] This pull request addresses an issue that has been marked with the 'Pull request accepted' label & I have added the link to that issue.
- [x] It contains one or more small, incremental, logically separate commits, with no merge commits.
- [x] I have used clear commit messages.
- [ ] Existing tests pass for me locally & I have added or updated tests for new or changed functionality.
- [ ] The work herein is covered by a valid [Contributor License Agreement (CLA)](https://github.com/openlayers/cla).

### CODE REVIEW

**Comment by bjornharrtell** [comment]
Replaces #4668. Trivial addition but should be unit tested, I realize now due to the nice reminders in the new PR template.

File: externs/olx.js [hunk 1]
```diff
@@ -3135,6 +3135,16 @@ olx.interaction.DrawOptions.prototype.geometryName;
...
+/**
+ * A function that takes an {@link ol.MapBrowserEvent} and returns a boolean
+ * to indicate whether the current sketch being drawn should be updated. This
+ * can be used to either limit the scope of the feature being drawn, or to
+ * skip events caused by panning/zooming on a touch device. By default
+ * {@link ol.events.condition.always}, i.e. it is always updated.
+ * @type {ol.EventsConditionType|undefined}
+ * @api
+ */
+olx.interaction.DrawOptions.prototype.updateSketchCondition;
```
Comment (fredj) [inline externs/olx.js]: The new option must also be added in the `olx.interaction.DrawOptions` `@typedef` above.

**Comment by bjornharrtell** [comment]
This PR is a suggested fix for the issue https://github.com/openlayers/openlayers/issues/3907.

File: externs/olx.js (hunk 1 above)
Comment (bjornharrtell) [reply externs/olx.js]: Thanks for noticing, fixed by https://github.com/openlayers/openlayers/pull/7535/commits/b53d61d9ff8586813cf9f88945d39c17b1c24b13.

**Comment by bjornharrtell** [comment]
I'm not sure CLA was established for @daanoz. However, the changes are quite trivial.

**Comment by ahocevar** [comment]
Good point. @daanoz, can you please sign a [CLA](https://github.com/openlayers/cla)? Thanks!

**Comment by daanoz** [comment]
Hi, thx for the interest in closing this PR. Since I've submitted this PR (and it's fix) as part of an enterprise application, I also rather get the CLA signed by the enterprise rather than me personally signing it. But, that has proven to be quite the challenge, the related ticket has been on our own scrumboard for over an year now, so I can imagine the pain with having this ticket open for so long. Therefore I propose to just close this ticket and the branch, and see if someone else is willing to pick it up, otherwise we're stuck in this deadlock?

**Comment by bjornharrtell** [comment]
I would like to pick it up and start from scratch, but it will likely result in mostly identical implementation - not sure if that's ok or not. Also, I've tried to make unit tests but got stuck on the event simulation, so I'll need some more time on it. Not expecting to get get it done until after 4.6 and so will close this.

**Pull request closed without merging**